SYCL/USM/allocator_rebind.cpp

```diff
@@ -0,0 +1,100 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t1.out
+// RUN: %HOST_RUN_PLACEHOLDER %t1.out
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+// RUN: %GPU_RUN_PLACEHOLDER %t1.out
+// RUN: %ACC_RUN_PLACEHOLDER %t1.out
+
+//==------------------------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <sycl.hpp>
+
+#include <iostream>
+
+using namespace sycl;
+
+// Allocator has a minimum internal alignment of 64, so use higher values for
+// the purpose of this test.
+struct alignas(256) Aligned256 {
+  int x;
+};
+struct alignas(128) Aligned128 {
+  int x;
+};
+
+template <class Allocator>
+bool is_overaligned(Allocator alloc, size_t Align,
+                    int Line = __builtin_LINE()) {
+  decltype(alloc.allocate(1)) Ptrs[10];
+  for (auto *&Elem : Ptrs)
+    Elem = alloc.allocate(1);
+
+  int NumExactlyAligned = 0;
+
+  for (auto *Ptr : Ptrs) {
+    auto Val = reinterpret_cast<uintptr_t>(Ptr);
```

Comment:
maybe rename `Val` to `Addr`  ? that might be clearer

Reply:
I didn't want to do that because its type wouldn't be a pointer. Can rename if you insist though.

Reply:
I wouldn't insist, but maybe persuade?  
If someone isn't reading this closely, then they'll likely conclude that `Val` is just the value held by the pointer. But clearly, we are testing the actual address of the pointer to ensure it's aligned. So I think `Addr`, or similar,  is clearer for the upcoming operations. Also, it makes people pay attention.  "Wow, he wants the address? That is so cool, I'd like to meet this guy"